example/w-nginx/nginx.conf

```diff
@@ -0,0 +1,34 @@
+
+user  nginx;
+worker_processes  1;
+
+error_log  /var/log/nginx/error.log notice;
+pid        /var/run/nginx.pid;
+
+
+events {
+    worker_connections  1024;
+}
+
+
+http {
+    include       /etc/nginx/mime.types;
+    default_type  application/octet-stream;
+
+    # Expose more information from the nginx logs.
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /dev/stdout;
+    error_log   /dev/stderr;
```

Comment:
Are these settings redundant with the ones in `default.conf`? Also, if we have total control of the server and are only serving one site, would it be better to merge the configs? Or is that considered a bad practice for some reason?

Reply:
The main difference from the default settings are lines 19-22 (changing the log format) and the include statement on line 33. The logs on the default nginx container are so terse that I found it hard to see what's going on.

I'm not an nginx expert by any means, but typically when I had to manage several different resources behind the reverse proxy, I did prefer to break those out into smaller files. Looking at [this discussion](https://serverfault.com/questions/527630/difference-in-sites-available-vs-sites-enabled-vs-conf-d-directories-nginx#527644), specifically the comments from `adamczi`, it looks like people do fold their site configuration into `nginx.conf` when they're deploying nginx in a container.

I think it depends a bit on whether the example is supposed to show how to use nginx in some generality (in which case, multiple configs are a bit more idiomatic), or if it instead should illustrate setting up a _containerized_ reverse proxy (in which case one config is better).

Reply:
I say to show a containerized version because...

- It is containerized in this example.
- Containerized is a pretty common way to do it nowadays.

There is a third reason to *strongly* prefer configuration in one file:

- The examples must be as concise as possible, subject to still being correct and informative. If two files are not necessary, and not even the norm for this particular kind of setup, there is no reason to pay the attention cost of the reader figuring out two `.conf`  files, trying to understand the relationship between them, etc.

Reply:
That makes sense. I've combined the two configs and made the combined config more concise.